distro/otel.go

```diff
@@ -26,7 +26,7 @@ import (
 	"go.opentelemetry.io/otel/sdk/metric"
 	"go.opentelemetry.io/otel/sdk/resource"
 	"go.opentelemetry.io/otel/sdk/trace"
-	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
```

Comment:
I decided to bump only semconv for resource like in upstream.

Reply:
Can you open an issue to track upgrading the rest of the repository?

Reply:
https://github.com/signalfx/splunk-otel-go/issues/2486